src/python/WMQuality/Emulators/RucioClient/MockRucioApi.py

```diff
@@ -58,9 +58,12 @@ def sitesByBlock(self, block):
         :return: a fake list of sites where the data is
         """
         logging.info("%s: Calling mock sitesByBlock", self.__class__.__name__)
-        if hash(block) % 3 == 0:
+        block = encodeUnicodeToBytesConditional(block, condition=PY3)
+        # this algorithm gives same results in both python versions
+        blockHash = int(hashlib.sha1(block).hexdigest()[:8], 16)
```

Comment:
This one I'm planning to leave with 8 chars slice only because otherwise there will be a bunch of tests that I will have to hunt down. Anyways, it's only in the mock module as well :-)

Reply:
I was about to ask you about this, and your motivation is sound! 
if the unittests pass and have no problem, then there is no need to change this and change all the unittests again